### PR TITLE
Npc list improvement

### PR DESCRIPTION
**Pull request by Firiman**
Added meet button functionality and confirmation before character deletion

### CODE REVIEW

File: UI/NpcList/NpcList.gd [hunk 1]
```diff
@@ -6,21 +6,95 @@ onready var nameButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Name
...
+	match occupation:
+		"Inmates": 
+			if(WorldPopulation.Inmates in GM.pc.getLocationPopulation()):
+				GM.main.runScene("InmateExposureForcedSexScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no inmates in this location.\nTry looking elswhere")
+		"Guards":
+			if(WorldPopulation.Guards in GM.pc.getLocationPopulation()):
+			#if(room.loctag_GuardsEncounter || room.loctag_Greenhouses):
```
Comment (Firiman) [inline UI/NpcList/NpcList.gd:58]: Which one is better, commented or WorldPopulation.Guards in GM.pc.getLocationPopulation()

File: UI/NpcList/NpcList.gd [hunk 2]
```diff
@@ -6,21 +6,95 @@ onready var nameButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Name
...
+
+func forgetNPC(ID, name, node):
+	_IDtoForget = ID
+	nodeToFree = node
+	sendPopupMessage("Are you sure you want to forget "+name+"? This will permanently remove "+name+" from the game. Keep in mind that if this character is pregnant, their pregnancy will be forgotten too. But any kids you had together will stay.", true)
+
+
+func _on_Forget_pressed():
+	nodeToFree.queue_free()
+	nodeToFree = null
```
Comment (Firiman) [inline UI/NpcList/NpcList.gd:38]: Made it so the node get no reference and die in peace. Is this correct?

File: UI/NpcList/NpcList.gd (hunk 1 above)
Comment (Alexofp) [reply UI/NpcList/NpcList.gd:58]: yours is fine

File: UI/NpcList/NpcList.gd [hunk 3]
```diff
@@ -6,21 +6,95 @@ onready var nameButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Name
...
+
 
-# add pop-up confirmation window
-func forgetNPC(ID):
-	GM.main.removeDynamicCharacter(ID)
+func hideNotificationButnsAndWindow():
+	popupForgetBtn.visible = false
+	popupOkBtn.visible = false
+	popupCancelBtn.visible = false
+	popupWindow.visible = false
```
Comment (Alexofp) [inline UI/NpcList/NpcList.gd:97]: hiding/showing only the popupWindow should be enough, the buttons are its childs so they will be hidden if the parent is hidden

File: UI/NpcList/NpcList.gd (hunk 2 above)
Comment (Alexofp) [reply UI/NpcList/NpcList.gd:38]: yep

File: UI/NpcList/NpcList.gd [hunk 4]
```diff
@@ -6,21 +6,95 @@ onready var nameButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Name
...
+				GM.main.runScene("EngineerCaughtOfflimitsScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no engineers in this location.\nTry looking at the engineering bay")
+		"Nurses":
+			if(room.loctag_MentalWard):
+				GM.main.runScene("NurseCaughtOfflimitsScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no nurses in this location.\nTry looking at the restricted area of the mediacl ward")
```
Comment (Alexofp) [inline UI/NpcList/NpcList.gd:74]: typo `mediacl`
`There are no nurses in this location.\nTry searching in the restricted area of the medical ward` would sound better

File: UI/NpcList/NpcList.gd [hunk 5]
```diff
@@ -6,21 +6,95 @@ onready var nameButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Name
...
+
+func meetNPC(ID, occupation):
+	var room = GM.world.getRoomByID(GM.pc.getLocation())
+	match occupation:
+		"Inmates": 
+			if(WorldPopulation.Inmates in GM.pc.getLocationPopulation()):
+				GM.main.runScene("InmateExposureForcedSexScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no inmates in this location.\nTry looking elswhere")
```
Comment (Alexofp) [inline UI/NpcList/NpcList.gd:55]: typo `elswhere`

File: UI/NpcList/NpcList.gd [hunk 6]
```diff
@@ -6,21 +6,95 @@ onready var nameButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Name
...
+
+
+func _on_Ok_pressed():
+	hideNotificationButnsAndWindow()
+
 
-# add pop-up confirmation window
-func forgetNPC(ID):
-	GM.main.removeDynamicCharacter(ID)
+func hideNotificationButnsAndWindow():
```
Comment (Alexofp) [inline UI/NpcList/NpcList.gd:93]: technically you don't need this function `popupWindow.visible = false` would be enough ^^. If you still want this function I'd suggest naming it `hideNotificationWindow`. Also don't shorten the words like `Butns`, looks better if you don't ^^

**Review by Alexofp — COMMENTED** [review]
I would probably need to write special scenes for when you meet the npcs through this window in the future, less agressive ones x3

**Comment by Alexofp** [comment]
Looks good, thankies ^^

**Pull request merged**